src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs

```diff
@@ -96,10 +97,16 @@ private static IDiagnoser[] LoadClassic()
 
         private static Assembly LoadDiagnosticsAssembly(Assembly benchmarkDotNetCoreAssembly)
         {
+            // We couldn't use Assembly.GetEntryAssembly()?.GetReferencedAssemblies() because
```

Comment:
I think that we should keep the old way (to make it work for old .NET frameworks) and extend it with the new one.

So it would be:

```cs
var referencedAssemblyName = Assembly.GetEntryAssembly()?.GetReferencedAssemblies().SingleOrDefault(name => name.Name == DiagnosticAssemblyName);	            if (referencedAssemblyName != default(AssemblyName))
    return Assembly.Load(referencedAssemblyName);

var inCompileLibraries = DependencyContext.Default.CompileLibraries.Any(l => l.Name.Equals(DiagnosticAssemblyName, StringComparison.OrdinalIgnoreCase));
var inRuntimeLibraries = DependencyContext.Default.RuntimeLibraries.Any(l => l.Name.Equals(DiagnosticAssemblyName, StringComparison.OrdinalIgnoreCase));

if (inCompileLibraries || inRuntimeLibraries)
    return Assembly.Load(new AssemblyName(DiagnosticAssemblyName));
```

Reply:
I'll check old .Net frameworks path.

Reply:
Agreed. `DependencyContext` doesn't work "by default" on .NET Framework. DependencyContext is mostly just for the .NET Core case.

You should also check for `DependencyContext.Default == null`, which will be the case when a `.deps.json` file isn't present.